### ISSUE TITLE
Helm does not allow release names to be re-used

### ISSUE THREAD

**Issue by maratoid** [kubernetes/helm#972]
Release names cannot be used again, after a named release has been deleted.

For example:
```
> helm install atlas/cluster-monitoring-0.1.0 --name monitoring
> helm delete monitoring
> helm install atlas/cluster-monitoring-0.1.0 --name monitoring
Error: a release named "monitoring" already exists
> helm delete monitoring
Error: [services "heapster" not found, deployments.extensions "monitoring-influxdb-grafana-v2" not found, services "monitoring-influxdb" not found, services "monitoring-grafana" not found, deployments.extensions "heapster-v10" not found]
```

This is a different error message that is generated for deletion of a release that has never been created before. 

From #helm channel chat, it seems that release names are retained forever so that helm can do delete rollbacks in the future - however some sort of a 'force' flag might be appropriate?

**Comment by technosophos** [comment]
I'm adding a check to change that second error to `the release named %q is already deleted`.

**Comment by technosophos** [comment]
This is dependent on #656 (because it needs the query api)

**Comment by technosophos** [comment]
#979 will fix the error message part, but the other part requires a fix that is being done as part of #656.
